### PR TITLE
2.1.0

### PR DESCRIPTION
**Pull request by hv-dev**
# API Pull Request
## VERSION Number
- 2.1.0
## Ticket url
- https://chillibean.atlassian.net/browse/PVA-798
## Changes Made
- Added accounts endpoint to schema
## Checklist
Please verify this checklist so that you are sure this is ready for review.

- [x]  The only file that has been changed is schema.json - unless you needed to add some new functionality (if so, please update generator repo too)
- [x]  Updated package.json & schema.json versions - they need to be the same
- [ ]  Verify there are no issues with the schema.json by running `npm run verify`
- [x]  All objects have a properites list, required array and additionalProperties set to false

### CODE REVIEW

File: public/refs/components/schemas/Accounts.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "type": "object",
```
Comment (damonspencerroberts) [inline public/refs/components/schemas/Accounts.json:2]: type array here

File: public/refs/components/schemas/Account.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (damonspencerroberts) [inline public/refs/components/schemas/Account.json:1]: Can you make this file BasicAccount

File: public/refs/components/schemas/Accounts.json (hunk 1 above)
Comment (damonspencerroberts) [reply public/refs/components/schemas/Accounts.json:2]: can you make this file BasicAccounts

File: public/refs/components/schemas/Accounts.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+  "items": {
+    "$ref": "./Account.json"
```
Comment (damonspencerroberts) [inline public/refs/components/schemas/Accounts.json:4]: `./BasicAccount.json`

File: public/refs/paths/accounts.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      "tags": [
+        "Accounts"
+      ],
+      "responses": {
+        "200": {
+          "description": "OK",
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "../components/schemas/Accounts.json"
```
Comment (damonspencerroberts) [inline public/refs/paths/accounts.json:25]: `../components/schemas/BasicAccounts.json`

File: public/schema.json [hunk 5]
```diff
@@ -52,6 +59,12 @@
       }
     },
     "schemas": {
+      "Account": {
```
Comment (damonspencerroberts) [inline public/schema.json:62]: Should have 3 here for accounts:
- BasicAccount
- BasicAccounts
- Account

Can you add the last one as a schema - that would belong in the `account/:id` endpoint - currently it is hard coded but would be better that it is defined in a schema

File: public/refs/components/schemas/Accounts.json (hunk 1 above)
Comment (hv-dev) [reply public/refs/components/schemas/Accounts.json:2]: making this an object is what allowed the tests to pass despite an array being returned.

File: public/fullSchema.json [hunk 6]
```diff
@@ -7,6 +7,10 @@
     "contact": {
       "name": "ChilliPharm",
       "url": "https://www.chillipharm.com"
+    },
+    "license": {
```
Comment (damonspencerroberts) [inline public/fullSchema.json:11]: You can remove the license - it's just a warning in the lint so no need to worry about it - just when it's an error do we act on it

**Review by damonspencerroberts — COMMENTED** [review]
Cool looks good just add the other endpoints in this branch from the 2 other tickets

File: public/refs/components/schemas/StudyArm.json [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    "account_id": {
+      "type": "number",
+      "description": "The id of the study arm",
+      "example": 1
+    },
+    "study_protocol_id": {
+      "type": "number",
+      "description": "The id of the study arm",
+      "example": 1
+    }
```
Comment (damonspencerroberts) [inline public/refs/components/schemas/StudyArm.json:24]: Missing subjects property:
full response looks something like:
```
{
    "id": 1,
    "display_name": "A",
    "created_at": "2024-08-12T09:59:25.493Z",
    "study_protocol_id": 2,
    "subjects": [
        {
            "id": 6,
            "subject_number": "B0001001",
            "created_at": "2024-08-12T10:00:22.168Z",
            "current_study_event": "REFILL1",
            "study_arm": "A",
            "created_by": "Mon Admin1",
            "creator": {
                "id": 16,
                "email": "chillitest53+monadmin1@gmail.com",
                "uuid": "db585b21-376f-4914-8089-fa1882a88d1b",
                "first_name": "Mon",
                "last_name": "Admin1"
            }
        }
    ]
}
```

Comment (damonspencerroberts) [reply public/refs/components/schemas/StudyArm.json:24]: Also no need for accountid

File: public/refs/components/schemas/StudyProcedureDefinition.json [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+      "type": "string",
+      "description": "The identifier of the study procedure definition",
+      "example": "IpsumC3H8F"
+    },
+    "position": {
+      "type": "number",
+      "description": "The position of the study procedure definition",
+      "example": 1
+    },
+    "required_form_templates": {
```
Comment (damonspencerroberts) [inline public/refs/components/schemas/StudyProcedureDefinition.json:25]: This is formatted incorrectly double check the full response below:
```
[
    {
        "date": "2024-08-12",
        "display_name": "Example Procedure",
        "evaluator_id": 16,
        "id": 3,
        "identifier": "EXPROC",
        "protocol": {
            "id": 2,
            "name": "Alpha Protocol",
            "study_event_display_name": {
                "plural": "refill visits",
                "singular": "refill visit"
            },
            "study_procedure_display_name": {
                "plural": "procedures",
                "singular": "procedure"
            },
            "locked": true,
            "version_number": 1,
            "study_procedure_role_definitions": []
        },
        "role_definitions": [],
        "status": "in_progress",
        "locked": false,
        "required_form_templates": [],
        "required_form_id": []
    },
    {
        "date": "2024-08-12",
        "display_name": "Final",
        "evaluator_id": 16,
        "id": 4,
        "identifier": "FINAL",
        "protocol": {
            "id": 2,
            "name": "Alpha Protocol",
            "study_event_display_name": {
                "plural": "refill visits",
                "singular": "refill visit"
            },
            "study_procedure_display_name": {
                "plural": "procedures",
                "singular": "procedure"
            },
            "locked": true,
            "version_number": 1,
            "study_procedure_role_definitions": []
        },
        "role_definitions": [],
        "status": "in_progress",
        "locked": false,
        "required_form_templates": [
            {
                "id": 1,
                "title": "Patient Review Form Template",
                "version_number": 1
            }
        ],
        "required_form_id": [
            3
        ]
    }
]
```

Comment (damonspencerroberts) [reply public/refs/components/schemas/StudyProcedureDefinition.json:25]: Also some missing attributes

**Review by damonspencerroberts — CHANGES_REQUESTED** [review]
Looking pretty good - just a few things to fix with the data.

File: public/schema.json [hunk 9]
```diff
@@ -34,6 +41,12 @@
     },
     "/accounts/{account_id}/form_templates/{id}": {
       "$ref": "refs/paths/accounts_{account_id}_form_templates_{id}.json"
+    },
+    "/accounts/{account_id}/protocols/{study_protocol_id}/study_arms": {
+      "$ref": "refs/paths/accounts_{account_id}_protocols_{study_protocol_id}_study_arms.json"
+    },
+    "/accounts/{account_id}/protocols/{study_protocol_id}/study_procedure_definitions": {
```
Comment (damonspencerroberts) [inline public/schema.json:48]: Sorry I made a mistake here it should be this response actually:
```
[
    {
        "id": 6,
        "display_name": "Example Procedure",
        "identifier": "EXPROC",
        "required_form_templates": []
    },
    {
        "id": 7,
        "display_name": "Final",
        "identifier": "FINAL",
        "required_form_templates": [
            {
                "id": 1,
                "title": "Patient Review Form Template",
                "version_number": 1
            }
        ]
    }
]
```

**Review by damonspencerroberts — CHANGES_REQUESTED** [review]

**Review by damonspencerroberts — APPROVED** [review]
looks great! Once you verify the tests pass we can merge

**Review by damonspencerroberts — APPROVED** [review]

**Pull request merged**